slack.go

```diff
@@ -32,6 +34,16 @@ func Slack(markdown string, options ...SlackConvertOptions) string {
 		gitHubURL = "https://github.com"
 	}
 
+	if len(opt.CustomRefPatterns) > 0 {
+		// using start and end patterns make sure we will not find match middle of the word etc.
+		startOfPattern := `(?P<THE_START_OF_MATCH>^|[\s\[\(])`
+		endOfPattern := `(?P<THE_END_OF_MATCH>$|[\s:\]\),.!])`
+		for pattern, replace := range opt.CustomRefPatterns {
+			re = regexp.MustCompile(fmt.Sprintf(`%s%s%s`, startOfPattern, pattern, endOfPattern))
```

Comment:
just a note: it will throw a panic if can;t compile.
So any app that uses this might have a heart attack :D 

Would suggest `re, err := regexp.Compile()` , log an error and return empty string back.

Reply:
tnx, updated :)